### PR TITLE
Add tip on custom flags

### PR DESCRIPTION
**Pull request by manshreck**

### CODE REVIEW

File: _posts/2019-05-10-totw-98.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+sidenav: side-nav-tips.html
+published: true
+permalink: tips/98
+type: markdown
+order: "098"
+---
+
+*by [Bradley White](mailto:bww@google.com)*
+
+NOTE: this tip contrasts legacy flags (of the form `DEFINE_flagname`)
```
Comment (tituswinters) [inline _posts/2019-05-10-totw-98.md]: legacy flags only exist as the package we open sourced a decade back (gflags)

File: _posts/2019-05-10-totw-98.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+with new-style Abseil flags of the form `ABSL_FLAG()`. The legacy forms
+are now deprecated; as a result, some of this tip is historical in
+nature, but the advice on using custom flags is current.)
+
+*"Do you have a flag?" - Eddie Izzard.*
+
+## Flags of User-Defined Types
+
+The Abseil Flags library now supports flags of user-defined types.
```
Comment (tituswinters) [inline _posts/2019-05-10-totw-98.md]:
```suggestion
The Abseil Flags library supports flags of user-defined types in addition to the types supported out-of-the-box.
```

File: _posts/2019-05-10-totw-98.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+`int32_t`, `int64_t`, `uint64_t`, `double`, `std::string`, where a flag
+definition like ...
+
+```c++
+DEFINE_string(default_host, "www.google.com", "For requests with no host");
+```
+
+expands directly into the definition of a global variable: `FLAGS_default_host`,
+of type `std::string`, with a default value of `"www.google.com"` and the given
+description. These flags are deprecated and people should now prefer the
```
Comment (tituswinters) [inline _posts/2019-05-10-totw-98.md]: I'm not sure we can say "deprecated" for an API in a project that we no longer maintain and has been forked ...

File: _posts/2019-05-10-totw-98.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+Consider using a flag of a user-defined type in cases where it is a clear gain
+in readability and safety.
+
+Flags specifying durations are an excellent example of where such a change would
+be beneficial, and `ABSL_FLAG(absl::Duration, ...)` is already available for
+your use. Remove legacy uses of ambiguous duration flags and the possibility of
+error by migrating your own duration flags to the new style.
+
+[parse-duration]: https://github.com/abseil/abseil-cpp/blob/bf29470384a101b307873b26d358433138c857fc/absl/time/time.h#L527
```
Comment (tituswinters) [inline _posts/2019-05-10-totw-98.md]: Are we sure this is the link we want for this?

**Comment by manshreck** [comment]
I've gone with not publishing this for now.

**Pull request merged**